tsl/src/reorder.c

```diff
@@ -148,12 +149,38 @@ tsl_move_chunk(PG_FUNCTION_ARGS)
 				 errmsg("valid chunk, destination_tablespace, and index_destination_tablespaces "
 						"are required")));
 
+	chunk = ts_chunk_get_by_relid(chunk_id, false);
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("\"%s\" is not a chunk", get_rel_name(chunk_id))));
+
+	if (ts_chunk_contains_compressed_data(chunk))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("\"%s\" is a compressed chunk", get_rel_name(chunk_id)),
+				 errhint("Please use the corresponding chunk on the uncompressed hypertable "
+						 "instead.")));
+
```

Comment:
The reorder will not do much when issued on the "main" chunk. I guess it will still move the empty chunk to the new tablespace. But I think this should be made clear in a comment and we should also consider whether we want to reorder the internal compressed chunk when possible. Maybe not for this PR, but in that case leave a comment on what is going on and that this is intentionally left out.